### PR TITLE
Turn on rest binary calculation tests

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] turning on Alonzo binary calculation golden test (here we have to adjust binary, as in alonzo we have isValid added to tx)
- [x] turning on Babbage binary calculation golden test (here we need to deal with cddl change for tx outputs in babbage -> https://github.com/input-output-hk/cardano-ledger/blob/74593d426d6f89cfc21d6f66549c5ad18e290177/eras/babbage/test-suite/cddl-files/babbage.cddl#L75 and hence to produce new binaries side by side with the binary from other eras ) 

### Issue Number

adp-1885

### CODE REVIEW

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs [hunk 1]
```diff
@@ -1551,33 +1614,66 @@ binaryCalculationsSpec' era = describe ("calculateBinary - "+||era||+"") $ do
...
+            calculateBinary net utxo outs chgs pairs `shouldBe` binary
 
   where
     slotNo = SlotNo 7750
     md = Nothing
     calculateBinary net utxo outs chgs pairs =
-        hex (Cardano.serialiseToCBOR ledgerTx)
+        case era of
+            ShelleyBasedEraAlonzo ->
+                slimCBOR eraSerializedCBOR
```
Comment (Anviking) [inline lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: Is it not saner to also add Alonzo-specific expected values in your case expressions above, like you already have for Babbage?

Comment (Anviking) [reply lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: In fact, the Alonzo and Babbage binaries are the same, no? I haven't run and tested, but it looks like https://github.com/input-output-hk/cardano-ledger/pull/2863 should have affected Alonzo as well.

Then having the same expected value for Alonzo and Babbage sounds much easier, and you could still keep your comment explaining the difference.

Comment (paweljakubas) [reply lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: thanks Johannes for your look. I tend to agree and now we have binary rep for Babbage, Alonzo and the rest. There are differences between all of them. I added the comment illustrating the point in the relevant test:
```
1421 -- Up till Mary era we have the following structure of transaction                                                                                                                      
1422 --   transaction =                                                                                                                                                                      
1423 -- [ transaction_body                                                                                                                                                                   
1424 -- , transaction_witness_set                                                                                                                                                            
1425 -- , auxiliary_data / null                                                                                                                                                              
1426 -- ]                                                                                                                                                                                    
1427 -- So we begin with 3-element array binary prefix, that is encoded as '83'                                                                                                              
1428 -- From Alonzo on tx was enriched for isValid field making it                                                                                                                           
1429 -- 4-element array that is encoded as '84'.                                                                                                                                             
1430 --   transaction =                                                                                                                                                                      
1431 -- [ transaction_body                                                                                                                                                                   
1432 -- , transaction_witness_set                                                                                                                                                            
1433 -- , bool                                                                                                                                                                               
1434 -- , auxiliary_data / null                                                                                                                                                              
1435 -- ]                                                                                                                                                                                    
1436 -- Alonzo transaction stil has the same binary representation (array)                                                                                                                   
1437 -- for transaction outputs like in the previous eras.                                                                                                                                   
1438 -- Babbage era changes this representation, and introduces map binary                                                                                                                   
1439 -- representation for transaction outputs as the contept of them is                                                                                                                     
1440 -- extended in this era.                                    
```

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs [hunk 2]
```diff
@@ -1414,12 +1414,30 @@ binaryCalculationsSpec (AnyCardanoEra era) =
...
+--   transaction =
+-- [ transaction_body
+-- , transaction_witness_set
+-- , bool
+-- , auxiliary_data / null
+-- ]
+-- Alonzo transaction stil has the same binary representation (array)
+-- for transaction outputs like in the previous eras.
+-- Babbage era changes this representation, and introduces map binary
+-- representation for transaction outputs as the contept of them is
```
Comment (Anviking) [inline lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]:
```suggestion
-- representation for transaction outputs as the concept of them is
```

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs (hunk 1 above)
Comment (Anviking) [reply lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: Ah ok so they are different after all 👌

**Review by Anviking — APPROVED** [review]

File: lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs (hunk 2 above)
Comment (paweljakubas) [reply lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs]: done

**Pull request merged**